crates/net/eth-wire/src/p2pstream.rs

```diff
@@ -209,26 +212,43 @@ pub struct P2PStream<S> {
     /// The supported capability for this stream.
     shared_capability: SharedCapability,
 
+    /// Messages coming from the swarm are multiplexed to the destined capability stream.
+    capability_mux:
+        HashMap<Capability, (mpsc::UnboundedSender<Bytes>, oneshot::Receiver<DisconnectReason>)>,
+
     /// Outgoing messages buffered for sending to the underlying stream.
-    outgoing_messages: VecDeque<Bytes>,
+    #[pin]
+    outgoing_messages: ReceiverStream<Bytes>,
+
+    /// Handle to pass to capabilities on their successful handshake.
+    outgoing_messages_handle: mpsc::Sender<Bytes>,
+
+    /// Handle to queue an outgoing ping message when the send buffer is full.
+    outgoing_pings_overflow: VecDeque<Bytes>,
 
     /// Whether this stream is currently in the process of disconnecting by sending a disconnect
     /// message.
     disconnecting: bool,
 }
 
 impl<S> P2PStream<S> {
```

Comment:
Is this type now still a stream?

because this type does the multiplexing now and delegates/receives messages from the connected capability streams.

I guess this now the actual Connection that wraps the connection stream/sink

Reply:
thanks, that's very valuable input, appreciate it